### PR TITLE
Fix #3543: compare extractor with TermRef instead of MethodType

### LINKED ISSUE #3543: Unreachable code warning for reachable match case
```scala
import scala.util.matching.Regex

object Test {
  def main(args: Array[String]): Unit = {
    foo("c" :: Nil, false)
  }

  def foo(remaining: List[String], inCodeBlock: Boolean): Unit = {
    remaining match {
      case CodeBlockEndRegex(before) :: ls =>
      case SymbolTagRegex(name) :: ls if !inCodeBlock => println("OK")
      case _ =>
    }
  }

  val CodeBlockEndRegex = new Regex("(b)")
  val SymbolTagRegex = new Regex("(c)")
}
```
Emits the folowing warining but should not.

```scala
11 |      case SymbolTagRegex(name) :: ls if !inCodeBlock => println("OK")
   |                                                      ^^^^^^^^^^^^^^^^
   |                                                      unreachable code
```

The code is correctly compiled and executed.

### PR DESCRIPTION
**Pull request by liufengyun**
Fix #3543: compare extractor with TermRef instead of MethodType

### CODE REVIEW

**Comment by allanrenucci** [comment]
If there is no check file, the tests expect that the compiler emit not warning?

File: tests/patmat/3543.scala [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    def unapply(x: String): Option[String] = ???
+  }
+
+  def test(xs: List[String]): Unit = {
+    val Yes = new Foo
+    val No = new Foo
+
+    xs match {
+      case Yes(x) :: ls => println("Yes")
+      case No(y) :: ls => println("No") // unreachable code
```
Comment (allanrenucci) [inline tests/patmat/3543.scala]: I would remove the comment

File: tests/patmat/3543.scala [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    def unapply(y: String): Boolean = x
+  }
+
+  def test(xs: List[String]): Unit = {
+    val Yes = new Foo(true)
+    val No = new Foo(false)
+
+    xs match {
+      case No() :: ls => println("No")
+      case Yes() :: ls => println("Yes") // unreachable code
```
Comment (allanrenucci) [inline tests/patmat/3543.scala]: I would remove the comment

**Comment by liufengyun** [comment]
@allanrenucci Yes, if there's no check file, there should be no warning.

**Pull request merged**